p2p/peer.py

```diff
@@ -485,6 +488,8 @@ def disconnect(self, reason: DisconnectReason) -> None:
         self.logger.debug("Disconnecting from remote peer; reason: %s", reason.name)
         self.base_protocol.send_disconnect(reason.value)
         self.close()
+        if self.is_running:
+            await self.cancel()
```

Comment:
cc @carver as you added a `wait_disconnect` method in your other PR and I think this is a better pattern.